src/librustc/lint/mod.rs

```diff
@@ -131,51 +131,104 @@ pub trait LintPass {
 // contains a few lint-specific methods with no equivalent in `Visitor`.
 pub trait LateLintPass: LintPass {
```

Comment:
I might just be failing at HKL, but I can't create a `Box<for<'a, 'tcx: 'a> LateLintPass<'a, 'tcx>>`, because the outlives bound is ignored

I have a workaround, but it'll require an intermediate helper trait. (see minimal example of workaround: https://is.gd/OnC7RS)